src/main/java/net/minecraftforge/resource/ResourceCacheManager.java

```diff
@@ -413,6 +446,178 @@ public boolean cacheLoaded()
+         * @return The current configuration file contents.
+         */
+        @NotNull
+        private static List<String> readConfigFileContent()
+        {
+            if (!Files.exists(CONFIG_PATH))
+            {
+                try
+                {
+                    Files.write(CONFIG_PATH, ImmutableList.of(
+                                    "# This TOML configuration file controls the resource caching system which is used before the mod loading environment starts.",
+                                    "# This file is read by the Forge boot loader, and is not used by the game itself.",
+                                    "#",
+                                    "# Set this to false to disable the resource cache. This will cause the game to scan the resource packs everytime it needs a list of resources.",
+                                    "cacheResources=true",
+                                    "",
+                                    "# Set this to false to forge the caching of vanilla resources to happen on the main thread.",
+                                    "indexVanillaPackCachesOnThread=false",
+                                    "",
+                                    "# Set this to false to forge the caching of mod resources to happen on the main thread.",
+                                    "indexModPackCachesOnThread=false"
+                            ),
+                            StandardOpenOption.CREATE_NEW);
+                }
+                catch (IOException e)
+                {
+                    throw new RuntimeException("Failed to create the default boot configuration file.", e);
+                }
+            }
+
+            final List<String> lines;
+            try
+            {
+                lines = Files.readAllLines(CONFIG_PATH);
+            }
+            catch (IOException e)
+            {
+                throw new RuntimeException("Failed to read the boot configuration file.", e);
+            }
+            return lines;
+        }
+
+        /**
+         * Reads a config value from the current configuration file.
+         *
+         * @param configKey    The key of the config value to read.
+         * @param defaultValue The default value to return if the config value is not present.
+         * @return The value of the config value, or the default value if it is not present.
+         */
+        private boolean getConfigValue(final String configKey, final boolean defaultValue)
+        {
+            //Get all lines.
+            final List<String> lines = getConfigFileContent();
+
+            for (final String line : lines)
+            {
+                //For each line: Remove all strings, and then check if it is the line with the requested key value.
+                final String trimmedLine = line.trim().replace(" ", "");
+                if (trimmedLine.startsWith("%s=".formatted(configKey))) //We look for key=value combindation in the trimmed
```

Comment:
According to the [TOML Spec](https://toml.io/en/v1.0.0#keyvalue-pair):
> Whitespace is ignored around key names and values.

Additionally, if as you said, the reading of the file happens few times, I do not see a problem with using nightconfig to read it.

Reply:
Misread by @Matyrobbrt Tested the base file in toml lint.
Works.